chapters/feature-selection.qmd

```diff
@@ -0,0 +1,525 @@
+
+The model with the smallest RMSE selected the top `r mlp_rf_best$top_p` most important predictors, `r mlp_rf_best$hidden_units` hidden units, a penalty value of 10<sup>`r signif(log10(mlp_rf_best$penalty), 3)`</sup>, a learning rate of 10<sup>`r signif(log10(mlp_rf_best$learn_rate), 3)`</sup>, and `r gsub("_", "-", mlp_rf_best$activation)` activation. The resampled RMSE associated with the numerically best results was `r round(show_best(mlp_rf_sfd_res, metric = "rmse", n = 1)$mean, 3)`. When that model was fit on the entire training set, the holdout RMSE estimate was `r round(mlp_rf_mtr$.estimate, 3)`.  One reason that these two estimates of performance are so close is the relatively high $n_{tr}/p$ ratio (1,000 / 29 $\approx$ 34.5).
+
+```{r}
+#| label: fig-mlp-rf
+#| echo: false
+#| warning: false
+#| out-width: 100%
+#| fig-width: 10
+#| fig-height: 5
+#| fig-align: center
+#| fig-cap: "Panel (a): The results of tuning a variance importance filter and the parameters from a single-layer neural network for predicting the simulated data from @eq-hooker.  Panel (b): The frequency distribution of predictor selection in the 10 folds for the model with the smallest RMSE."
+
+p_rmse <- autoplot(mlp_rf_sfd_res, metric = "rmse") +
+  ylim(c(.24, .72)) +
+  geom_hline(yintercept = 0.25, col = "darkgreen", lty = 2) +
+  labs(title = "(a) Resampled RMSE results")
+
+p_terms <- retained_variables %>%
+  mutate(
+    terms = gsub("predictor", "x", terms)
+  ) %>%
+  ggplot(aes(terms)) +
+  geom_bar() +
+  coord_flip() +
+  labs(
+    x = NULL,
+    y = "# Selections in 10 Folds",
+    title = "(b) Selection Frequency"
+  ) +
+  scale_y_continuous(breaks = pretty_breaks())
+
+(p_rmse | p_terms) + plot_layout(widths = c(4, 3))
+```
+
+Recall that the importance scores are computed on different analysis sets. For the best model, @fig-mlp-rf(b) contains the frequency at which each predictor was selected. For the most part, the importance scores are pretty accurate; almost all informative features are consistently selected across folds. When the final model was fit on the entire training set, `r cli::format_inline("{num_noise} noise predictor{?s} {?was/were}")`  included in the model along with predictors `r cli::format_inline("{non_noise}")`. These results are unusually clean, mostly due to the large sample size and relatively small predictor set.
```

Comment:
I see now that "1 noise predictor was included in the model" was computer generated. I feel like it would need a sentence pointing out why.

it almost read like it was a active choice to include one, but it should really be "our model wasn't able to remove this one"